ValheimPlus/GameClasses/EnvMan.cs

```diff
@@ -25,41 +25,91 @@ public static void SetupDayLength()
             }
         }
 
+        private const float startOfADay = 0.25f;
+        private const float startOfNight = 0.75f;
+        private const float startOfAfternoon = 0.5f;
+
+        /// <summary>
+        /// Return progression of the day on a range of 0 to 1
+        /// </summary>
+        public static double GetDayProgression()
+        {
+            double timeSecondsFromStart = ZNet.instance.GetTimeSeconds();
+            double timeSecondsFromDay = timeSecondsFromStart % Configuration.Current.Time.totalDayTimeInSeconds;
+            return timeSecondsFromDay / Configuration.Current.Time.totalDayTimeInSeconds;
+        }
+
         /// <summary>
         /// Hook on EnvMan init to alter total day length
         /// </summary>
         [HarmonyPatch(typeof(EnvMan), "Awake")]
-        public static class TimeInitHook
+        public static class EnvMan_Awake_Patch
         {
             private static void Prefix(ref EnvMan __instance)
             {
                 SetupDayLength();
             }
         }
 
+        [HarmonyPatch(typeof(EnvMan), "IsDay")]
+        class EnvMan_IsDay_Patch
```

Comment:
1. Can we add the class modifiers to follow (admittedly undocumented) convention? eg `public static` in front of `class`?
2. Could we add `///summary` above these new patches?

Reply:
will do